config.yml

```diff
@@ -328,6 +318,21 @@ tokens:
     comment: "a separator between words in a list"
   - name: __END__
     comment: "marker for the point in the file at which the parser should stop"
+  - name: KEYWORD_IF_MODIFIER
+    value: 32768
+    comment: "if in the modifier form"
+  - name: KEYWORD_UNLESS_MODIFIER
+    value: 32769
+    comment: "unless in the modifier form"
+  - name: KEYWORD_WHILE_MODIFIER
+    value: 32770
+    comment: "while in the modifier form"
+  - name: KEYWORD_UNTIL_MODIFIER
+    value: 32771
+    comment: "until in the modifier form"
+  - name: KEYWORD_RESCUE_MODIFIER
+    value: 32772
```

Comment:
Such high values means it's no longer possible to simply use an Array of token value -> token symbol or so anymore, notably this shows up in https://github.com/ruby/yarp/actions/runs/6314711880/job/17145622799?pr=1601.
A Hash would likely be much slower and use more memory.
So maybe some logic for the special cases >= 32768?

Reply:
Another idea is if the only flag we need (so far at least) for tokens is "statement vs expression" the check for "is it a statement token?" could be just "token >= KEYWORD_IF_MODIFIER", i.e., having the statement tokens at the end (and adding a comment in this file for where the limit is, or even an unused token to separate clearly).